pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -1171,15 +1172,19 @@ func (a *ApplicationRestoreController) restoreResources(
 	if err := a.applyResources(restore, objects); err != nil {
 		return err
 	}
-
+	// Before  updating to final stage, cleanup generic backup CRs, if any.
+	err = a.cleanupResources(restore)
+	if err != nil {
+		return err
+	}
 	restore.Status.Stage = storkapi.ApplicationRestoreStageFinal
 	restore.Status.FinishTimestamp = metav1.Now()
 	restore.Status.Status = storkapi.ApplicationRestoreStatusSuccessful
 	restore.Status.Reason = "Volumes and resources were restored up successfully"
 	for _, resource := range restore.Status.Resources {
 		if resource.Status != storkapi.ApplicationRestoreStatusSuccessful {
```

Comment:
If the overall status is Failure, wouldn't the reason say success with this message?